docs/adr/0000-whats-the-big-idea.md

```diff
@@ -0,0 +1,151 @@
+bindings to our rust code, and rely on the portable nature of WebAssembly to run on multiple platforms.
+
+* Good, because the tooling around `wasm-bindgen` seems quite sophisticated and mature.
+* Good, because this toolchain is maintained by folks for whom it is a full-time job.
+* Good, because it meshes well with technical projects that are strategically important for Mozilla.
+
+* Bad, because `wasm-bindgen` currently only supports JavaScript as a target language.
+* Bad, because we have not been able to identify a mature solution for running WebAssembly on Android or iOS,
+  which are important target platforms.
+* Bad, because this is a significant departure from how we've written components in the past, which adds
+  timeline risk to shipping this solution.
+
+Ultimately, we can imagine a world in which the WebAssembly ecosystem is sufficently advanced to
+make this the most compelling option, but that world seems far enough away that it doesn't make
+sense for us to pursue this right now.
+
+### Option C: Use SWIG, Djinni, or another existing bindings-generator tool
+
+Writing code in a systems language and generating bindings for high-level languages is not a new idea.
+Among existing tools in this general space are [SWIG](http://www.swig.org/) and
+[Djinni](https://github.com/dropbox/djinni). We could adopt one of these existing tools instead
+of inventing our own thing.
+
+* Good, because these tools already exist and are mature, saving us development and maintenance effort.
+* Good, because it would realize the goal of avoiding hand-written boilerplate.
+* SWIG:
+  * Bad, because it doesn't appear to have support for generating Kotlin or Swift bindings, which
+    are key languages for our use-case.
+  * Bad, because it is designed for C/C++ rather than Rust, meaning an unknown amount of exploratory
+    work required to integrate it with our approach before we can ship anything.
+* Djinni:
+  * Good, because it targets several of our key languages/platforms.
+  * Bad, because it is designed using C++ as the implementation language rather than Rust, , meaning
+    an unknown amount of exploratory work required to integrate it with our approach before we can
+    ship anything.
+  * Bad, because it has explicitly been put into "maintenance mode" by its authors.
+
+Ultimately, while we could probably make one of these tools work, we could not find one that was
+a close enough fit for our needs to avoid the "unknown amount of exploratory integration work" problem.
+We're not willing to put off shipping incremental progress for long enough to be confident of
+making the integration work.
+
+### Option D: Write and maintain a custom tool that automates our current best practices
+
+We can take the patterns we've established for writting the FFI layer and foreign language bindings
+by hand, and encapsulate them in a custom tool to automatically generate similar code.
+
```

Comment:
You've called it out a couple of times already, but also good—it has to supports all our target platforms and their requirements (Megazording, existing interface for calling between Rust and platform code, integration into Firefox Desktop) out of the box. The flip side is the bad you noted about a one-size-fits-all feature set, but designing the tool from the beginning to support our workflows means avoiding interface and integration churn.

Reply:
I'll add an additional bullet-point to call this out more explicitly 👍